pkg/controller/logstorage/logstorage_controller.go

```diff
@@ -660,51 +655,71 @@ func (r *ReconcileLogStorage) deleteInvalidECKManagedPublicCertSecret(ctx contex
 //    created.
 // 2) The certificate secret needed for Elasticsearch (in the Elasticsearch namespace).
 // 3) The certificate secret created by the ECK operator using the secret in 2).
-func (r *ReconcileLogStorage) getElasticsearchCertificateSecrets(ctx context.Context) (*corev1.Secret, *corev1.Secret, *corev1.Secret, error) {
+func (r *ReconcileLogStorage) getElasticsearchCertificateSecrets(ctx context.Context, instl *operatorv1.InstallationSpec) (oprKeyCert *corev1.Secret, esKeyCert *corev1.Secret, oprCert *corev1.Secret, err error) {
 	svcDNSNames := dns.GetServiceDNSNames(render.ElasticsearchServiceName, render.ElasticsearchNamespace, r.clusterDomain)
 
 	// Get the secret - might be nil
-	certSecret, err := utils.GetSecret(ctx, r.client, render.TigeraElasticsearchCertSecret, rmeta.OperatorNamespace())
+	oprKeyCert, err = utils.GetSecret(ctx, r.client, render.TigeraElasticsearchCertSecret, rmeta.OperatorNamespace())
 	if err != nil {
 		return nil, nil, nil, err
 	}
 
 	// Ensure that cert is valid.
-	certSecret, err = utils.EnsureCertificateSecret(render.TigeraElasticsearchCertSecret, certSecret, "tls.key", "tls.crt", rmeta.DefaultCertificateDuration, svcDNSNames...)
+	oprKeyCert, err = utils.EnsureCertificateSecret(render.TigeraElasticsearchCertSecret, oprKeyCert, corev1.TLSPrivateKeyKey, corev1.TLSCertKey, rmeta.DefaultCertificateDuration, svcDNSNames...)
 	if err != nil {
 		return nil, nil, nil, err
 	}
 
-	certSecretESCopy := rsecret.CopyToNamespace(render.ElasticsearchNamespace, certSecret)[0]
-
-	// Get the pub secret - might be nil
-	pubSecret, err := utils.GetSecret(ctx, r.client, relasticsearch.PublicCertSecret, render.ElasticsearchNamespace)
+	oprKeyCertIssuer, err := utils.GetCertificateIssuer(oprKeyCert.Data[corev1.TLSCertKey])
 	if err != nil {
 		return nil, nil, nil, err
 	}
 
-	// If the provided certificate secret (secret) is managed by the operator we need to check if the secret that
-	// Elasticsearch creates from that given secret (pubSecret) has the expected DNS name. If it doesn't, delete the
-	// public secret so it can get recreated.
-	if pubSecret != nil {
-		operatorManaged, err := utils.IsOperatorManaged(certSecret, "tls.crt")
+	// If Certificate management is enabled, we only want to trust the CA cert and let private key generation being handled from an init container.
```

Comment:
Nit "let private key generation being handled from an init container" => "let the init container handle private key generation"